internal/service/temporary_credentials.go

```diff
@@ -76,9 +77,54 @@ func (s *Store) Retrieve(_ context.Context, user common.Address) (*Credential, e
 		return nil, fmt.Errorf("failed to unmarshal credentials: %w", err)
 	}
 
-	if cred.AccessToken == "" || cred.RefreshToken == "" || cred.Expiry.IsZero() {
+	if cred.AccessToken == "" || cred.RefreshToken == "" || cred.AccessExpiry.IsZero() || cred.RefreshExpiry.IsZero() {
 		return nil, errors.New("credential was missing a required field")
 	}
 
 	return &cred, nil
 }
+
+func (s *Store) RetrieveWithTokensEncrypted(_ context.Context, user common.Address) (*Credential, error) {
+	cacheKey := prefix + user.Hex()
+	cachedCred, ok := s.Cache.Get(cacheKey)
+	if !ok {
+		return nil, ErrNotFound
+	}
+
+	encCred := cachedCred.(string)
+
+	// Don't want a second call to pick this up. Use it or lose it.
+	s.Cache.Delete(cacheKey)
```

Comment:
What second call when you talk about? Seems strange to me to delete, maybe set ttl?